test/hotspot/jtreg/gc/z/unloading/TestUnloadGarbageCollectorMXBean.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021, Alibaba Group Holding Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test TestUnloadGarbageCollectorMXBean
+ * @requires vm.gc.Z & !vm.graal.enabled
+ * @summary Test whether ZGC garbage collector MXBean reports pause time and count of class unloading
+ * @library / /test/lib /runtime/testlibrary
+ * @build sun.hotspot.WhiteBox
+ * @run driver ClassFileInstaller sun.hotspot.WhiteBox
+ *                                sun.hotspot.WhiteBox$WhiteBoxPermission
+ * @run main/othervm -XX:+UseZGC -XX:+ClassUnloading -XX:ZUnloadClassesFrequency=2 -Xms128m -Xmx128m
+ *      -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *      gc.z.unloading.TestUnloadGarbageCollectorMXBean 2
+ * @run main/othervm -XX:+UseZGC -XX:+ClassUnloading -XX:ZUnloadClassesFrequency=13 -Xms128m -Xmx128m
+ *      -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *      gc.z.unloading.TestUnloadGarbageCollectorMXBean 13
+ * @run main/othervm -XX:+UseZGC -XX:-ClassUnloading -Xms128m -Xmx128m
+ *      -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *      gc.z.unloading.TestUnloadGarbageCollectorMXBean 0
+ */
+package gc.z.unloading;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import sun.hotspot.WhiteBox;
+
+public class TestUnloadGarbageCollectorMXBean {
+    private static final WhiteBox wb = WhiteBox.getWhiteBox();
+    private static final long triggeredCycles = 150;
+    private static long unloadClassesFrequency;
+    private static long unloadClassesPauses;
+    private static long reportedCycles = 0;
+    private static long reportedPauses = 0;
+
+    public static void main(String[] args) throws Exception {
+        unloadClassesFrequency = Long.parseLong(args[0]);
+
+        // trigger gc
+        for (int i = 0; i < triggeredCycles; i++) {
+            wb.youngGC();
+        }
+
+        collectStatistics();
```

Comment:
is it possible to move `collectStatistics();` and `verify();` into the `for` loop in order to verify each GC cycle?

Reply:
done